nitrile-butadiene rubber

aliu-here opened the pull request:
## What
adding nitrile-butadiene rubber; based off Rubber, 3. Synthetic 2.3.2 in ullmann's encyclopedia of industrial chemistry, and wikipedia page [Radical polymerization](https://en.wikipedia.org/wiki/Radical_polymerization)

## Implementation Details
adds recipes for: nitrile-butadiene rubber & sodium benzylsulfonate detergent

## Outcome
adds nitrile rubber

## Additional Information
this is my first PR so i've probably messed up somewhere big time; also i don't know if coagulator should be used, but nitrile-butadiene rubber does need to be coagulated in a tank; i'm also kind of guessing for the amounts since ratios aren't specified apart from an approximate 2:1 ratio of butadiene : acrylonitrile

aliu-here committed:
nitrile rubber

aliu-here committed:
edit further

aliu-here committed:
fix typos, ratios

aliu-here committed:
oops i did git rm wrong

aliu-here committed:
forgot to pull

aliu-here committed:
fix format

aliu-here committed:
cleaning up my mess

aliu-here committed:
make surfactant name consistent, add missing materials

aliu-here committed:
fix susy id

aliu-here committed:
reorder, also fix typo

aliu-here commented:
<img width="1511" height="900" alt="image" src="https://github.com/user-attachments/assets/be9cb6ed-5fea-49e9-8aed-b097a42603b8" />
relevant excerpt

aliu-here committed:
add names

aliu-here committed:
fix names, should be benz**ene**sulfonate not benzylsulfonate

bruberu commented on groovy/postInit/chemistry/organic_chemistry/polymers/thermosets/RubberChain.groovy:
```diff
@@ -309,3 +309,61 @@ MIXER.recipeBuilder()
+
+// nitrile-butadiene rubber
+
+POLYMERIZATION_TANK.recipeBuilder()
```
This recipe just wouldn't work, since the polymerization tank only accepts up to three fluid inputs.

bruberu reviewed (changes requested):
I did find this one issue that would prevent GroovyScript from fully loading. I recommend trying to run in this in the modpack to look for more errors like this.

aliu-here committed:
split up polymerization tank reaction up into fewer reactants
also fix some typos

bruberu committed:
Merge remote-tracking branch 'origin/master-ceu' into nbr

bruberu committed:
fix: all the things

bruberu commented:
@aliu-here You might want to look at my latest commit, as there were some things that I did improve about the PR to help it run properly. It also still seems like you haven't run this in-game yet. If you want a demo of that, please contact me on Discord.
Also, I would recommend looking at CONTRIBUTING.md in the future (especially at our stoichiometric conventions).

planetme committed:
Merge branch 'master-ceu' into pr/1756

planetme committed:
fix: change NBR chain to be more accurate

Reworks the nitrile rubber chain:
- Emulsifier is changed to SDS (already in game)
- Changes redox initiator system to use an organic peroxide
- Sodium dithionite shortstopper
- Changed coagulant to be salt + acid
- Added vulcanization recipes for NBR

planetme committed:
fix: reorder DT outputs to be based on bp

Fiery36 commented on groovy/postInit/chemistry/organic_chemistry/polymers/thermosets/RubberChain.groovy:
```diff
@@ -74,7 +74,8 @@ def rubbers = [
     new Rubber('dustCompoundedStyreneIsopreneRubber', 'StyreneIsopreneRubber', 4, 30 * 20, 4),
-    new Rubber('dustCompoundedStyreneButadieneRubber', 'StyreneButadieneRubber', 4, 30 * 20, 4)
+    new Rubber('dustCompoundedStyreneButadieneRubber', 'StyreneButadieneRubber', 4, 30 * 20, 4),
+    new Rubber('dustRawNitrileRubber', 'NitrileButadieneRubber', 4, 30 * 20, 4)
```
The material is NitrileRubber not NitrileButadieneRubber

planetme committed:
fix bugs

trainvoi committed:
fixing bug

trainvoi committed:
Merge branch 'nbr' of https://github.com/aliu-here/Supersymmetry into pr/1756

planetme committed:
Revert "fixing bug"

This reverts commit ab8f5a4140ba23aebe673e1dcde492e67c8b2848.

planetme committed:
Merge branch 'master-ceu' into pr/1756

bruberu committed:
Merge branch 'master-ceu' into nbr
